overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -69,20 +69,32 @@ func getServiceInfos(st *state.State, snapName string, serviceNames []string) ([
 
 var servicestateControl = servicestate.Control
 
-func queueCommand(context *hookstate.Context, ts *state.TaskSet) {
-	// queue command task after all existing tasks of the hook's change
+func queueCommand(context *hookstate.Context, ts *state.TaskSet) error {
+	hookTask, ok := context.Task()
+	if !ok {
+		return fmt.Errorf("attempted to queue command with ephemeral context")
+	}
+
 	st := context.State()
 	st.Lock()
 	defer st.Unlock()
 
-	task, ok := context.Task()
-	if !ok {
-		panic("attempted to queue command with ephemeral context")
+	change := hookTask.Change()
+	hookTaskLanes := hookTask.Lanes()
+	tasks := change.LaneTasks(hookTaskLanes...)
+	if len(hookTaskLanes) == 1 && hookTaskLanes[0] != 0 {
```

Comment:
the if/else seems wrong, if the lanes are {0}  we don't need/want to join anything, maybe something like:

    if len(hookTaskLanes) == 1 && hooTaskLanes[0] == 0 {
       hookTaskLanes := nil
    }

    for _, l := range hookTaskLanes{
    ...

Reply:
we probably need a test for the no lanes aka lanes {0} case